site/app/controllers/AuthenticationController.php

```diff
@@ -308,4 +310,274 @@ public function userSelection() {
+        $confirm_password = $_POST['confirm_password'];
+
+        $verified_users = $this->core->getQueries()->getUserIdEmailExists($email, $user_id);
+        $unverified_users = $this->core->getQueries()->getUnverifiedUserIdEmailExists($email, $user_id);
+
+        if (in_array($email, array_column($unverified_users, 'user_email'), true) || in_array($email, array_column($verified_users, 'user_email'), true)) {
+            $this->core->addErrorMessage('Email already exists.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'email_verification']));
+        }
+
+        if (in_array($user_id, array_column($verified_users, 'user_id'), true)) {
+            $this->core->addErrorMessage('User ID already exists');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+
+        if (!$this->isGoodPassword($password)) {
+            $this->core->addErrorMessage('Password does not meet the requirements.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+
+        if ($password !== $confirm_password) {
+            $this->core->addErrorMessage('Passwords did not match.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+
+        if (!$this->isAcceptedEmail($email)) {
+            $this->core->addErrorMessage('This email is not accepted.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+
+        if (!$this->isAcceptedUserId($user_id, $_POST['given_name'], $_POST['family_name'], $email)) {
+            $this->core->addErrorMessage('This user id does not meet requirements.');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'create_account']));
+        }
+        $verification_values = $this->generateVerificationCode();
+        $user = new User($this->core, [
+            'user_id' => $user_id,
+            'user_givenname' => $_POST['given_name'],
+            'user_familyname' => $_POST['family_name'],
+            'user_password' => $password,
+            'user_pronouns' => '',
+            'display_pronouns' => false,
+            'user_email' => $email,
+            'user_email_secondary' => '',
+            'user_email_secondary_notify' => false,
+            'user_verification_code' => $verification_values['code'],
+            'user_verification_expiration' => $verification_values['exp']
+        ]);
+
+        try {
+            $this->core->getQueries()->insertUnverifiedSubmittyUser($user);
+            $this->sendVerificationEmail($email, $verification_values['code'], $user_id);
+            $this->core->addSuccessMessage('Verification Email Sent');
+            return new RedirectResponse($this->core->buildUrl(['authentication', 'email_verification']));
+        }
+        catch (\Error $e) {
```

Comment:
Is there something more specific we can catch here?

Reply:
I haven't tested it enough to figure out what would happen here, I just had it as a safety measure. I will test it more later next week to see.